internal/utils/cloudinit.go

```diff
@@ -0,0 +1,136 @@
+package utils
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+
+	"github.com/hashicorp/go-multierror"
+	"github.com/kairos-io/immucore/internal/constants"
+	"github.com/mudler/yip/pkg/console"
+	"github.com/mudler/yip/pkg/executor"
+	"github.com/mudler/yip/pkg/logger"
+	"github.com/mudler/yip/pkg/plugins"
+	"github.com/mudler/yip/pkg/schema"
+	"github.com/sirupsen/logrus"
+	"github.com/twpayne/go-vfs"
+	"gopkg.in/yaml.v3"
+)
+
+func NewYipExecutor(l logger.Interface) executor.Executor {
+	exec := executor.NewExecutor(
+		executor.WithLogger(l),
+		executor.WithConditionals(
+			plugins.NodeConditional,
+			plugins.IfConditional,
+		),
+		executor.WithPlugins(
+			// Note, the plugin execution order depends on the order passed here
+			plugins.DNS,
+			plugins.Download,
+			plugins.Git,
+			plugins.Entities,
+			plugins.EnsureDirectories,
+			plugins.EnsureFiles,
+			plugins.Commands,
+			plugins.DeleteEntities,
+			plugins.Hostname,
+			plugins.Sysctl,
+			plugins.User,
+			plugins.SSH,
+			plugins.LoadModules,
+			plugins.Timesyncd,
+			plugins.Systemctl,
+			plugins.Environment,
+			plugins.SystemdFirstboot,
+			plugins.DataSources,
+			plugins.Layout,
```

Comment:
wondering about the layout plugin here, shall we just have it in `yip`?

Reply:
It comes from yip directly, and should expand the partitions and all that crap. Not sure why elemental did have the plugin separated, if it was to inject the fs and runners and such or what. Will have a look and potentially send patches to yip, but as far as I can see, the bundled plugin in yip should expand the partitions correctly

Reply:
IIRC it was due to sgdisk/parted, no? IIRC the yip standard plugin supported only BIOS, not GPT disks

Reply:
yep, it used indeed the elemental-cli partitioner instead: https://github.com/kairos-io/elemental-cli/blob/04802f037647560f4d8a4a4f382ec1cb375f9d72/pkg/cloudinit/layout_plugin.go#L36

Reply:
also.... I remember patching some parts too in our fork

Reply:
argh, that sucks. Either we bundle a custom version here or we sent the patches to yip